src/directives/offense/harass.ts

```diff
@@ -0,0 +1,90 @@
+import {RoomIntel} from '../../intel/RoomIntel';
+import {profile} from '../../profiler/decorator';
+import {Directive} from '../Directive';
+import {HarassOverlord} from "../../overlords/offense/harass";
+import {log, Log} from "../../console/log";
+import {MY_USERNAME} from "../../~settings";
+
+interface DirectiveHarassMemory extends FlagMemory {
+	enhanced?: boolean;
+	targetPlayer?: string;
+	roomsToHarass: string[];
+}
+
+/**
+ * Harass Directive that wanders through enemy rooms killing stuff
+ * Placed on an enemy room, it will harass all of it's remotes periodically
+ */
+@profile
+export class DirectiveHarass extends Directive {
+
+	static directiveName = 'harass';
+	static color = COLOR_RED;
+	static secondaryColor = COLOR_WHITE;
+
+	memory: DirectiveHarassMemory;
+
+	constructor(flag: Flag) {
+		super(flag);
+		this.memory.targetPlayer = RoomIntel.roomOwnedBy(flag.pos.roomName);
+		if (this.memory.targetPlayer == MY_USERNAME) {
+			log.error(`Ahhhhhh harassing self in room ${flag.pos.roomName}`);
+			this.remove();
+		}
+		if (this.memory.targetPlayer) {
+			this.memory.roomsToHarass = this.findNearbyReservedRooms(flag.pos.roomName, this.memory.targetPlayer);
+		}
+	}
+
+	spawnMoarOverlords() {
+		// For now, just spawn from RCL 5+ rooms
+		this.overlords.harassOverlord = new HarassOverlord(this, this.memory.enhanced);
+	}
+
+	init(): void {
+		// if
+		// if (!this.memory.roomsToHarass && this.memory.targetPlayer)
+
+
+	}
+
+	findNearbyReservedRoomsForHarassment() {
+		if (this.memory.targetPlayer) {
```

Comment:
might make more sense to have `this.memory.targetPlayer` be a `string` rather than `string | undefined` and just remove the directive if no target player is found